contracts/cw3-multisig/src/state.rs

```diff
@@ -106,25 +106,22 @@ impl Proposal {
         self.status = self.current_status(block);
     }
 
-    /// Helper function to check if a certain vote count has reached threshold.
-    /// Only called from is_rejected and is_passed for no and yes votes
-    /// Handles the different threshold types accordingly.
-    /// This function returns true if and only if vote_count is greater than the threshold which
-    /// is calculated.
-    /// In the case where we use yes votes, this function will return true if and only if the
-    /// proposal will pass.
-    /// In the case where we use no votes, this function will return true if and only if the
-    /// proposal will be rejected regardless of other votes.
-    fn does_vote_count_reach_threshold(&self, vote_count: u64, block: &BlockInfo) -> bool {
+    /// Returns true if this proposal is sure to pass (even before expiration, if no future
+    /// sequence of possible votes could cause it to fail).
+    pub fn is_passed(&self, block: &BlockInfo) -> bool {
         match self.threshold {
             Threshold::AbsoluteCount {
                 weight: weight_needed,
-            } => vote_count >= weight_needed,
+            } => self.votes.yes >= weight_needed,
             Threshold::AbsolutePercentage {
                 percentage: percentage_needed,
             } => {
-                vote_count
-                    >= votes_needed(self.total_weight - self.votes.abstain, percentage_needed)
+                let opinions = self.total_weight - self.votes.abstain;
+                // If all abstains we cannot pass the propsal
+                if opinions == 0 {
```

Comment:
is this safe, we used Uint128::zero last time, why not here?

Reply:
types are different multisig uses u64